### PR TITLE
NTLM Profile section added

### PR DESCRIPTION
**Pull request by bmvkrd**

### CODE REVIEW

**Comment by bmvkrd** [comment]
Huilin, please review the change

File: on-prem.md [hunk 1]
```diff
@@ -270,6 +270,28 @@ kafka:
...
+```
+
+The following properties are supported:
+
+| Property name | Description |
+|------------------|-------------------------------------------|
+| auth | NTLM authentication credentials |
+| base_url | The base URL for NTLM resources |
+| cm_default_max_per_route | (Optional) Sets the number of connections per route (default 5) |
+| cm_max_total | (Optional) Sets the maximum number of connections (default 10) |
```
Comment (eeshansim) [inline on-prem.md]: Is there a max or min number?

File: on-prem.md [hunk 2]
```diff
@@ -270,6 +270,28 @@ kafka:
...
+    cm_max_total: 100
+```
+
+The following properties are supported:
+
+| Property name | Description |
+|------------------|-------------------------------------------|
+| auth | NTLM authentication credentials |
+| base_url | The base URL for NTLM resources |
+| cm_default_max_per_route | (Optional) Sets the number of connections per route (default 5) |
```
Comment (eeshansim) [inline on-prem.md]: Is there a max or min number?

Comment (eeshansim) [reply on-prem.md]: Also, will be good be to define what our definition of a route is.

File: on-prem.md (hunk 1 above)
Comment (bmvkrd) [reply on-prem.md]: Yep, set to positive

File: on-prem.md (hunk 2 above)
Comment (bmvkrd) [reply on-prem.md]: There is no "our" definition of a route, it is defined by HttpClient library, and quite complicated for the end users. Check "HTTP connection routing" section in this document: https://hc.apache.org/httpcomponents-client-ga/tutorial/html/connmgmt.html

**Pull request merged**